gems/aws-sdk-core/lib/aws-sdk-core/assume_role_web_identity_credentials.rb

```diff
@@ -16,12 +15,20 @@ module Aws
   #       role_session_name: "session-name"
   #       ...
   #     )
-  #     For full list of parameters accepted
-  #     @see Aws::STS::Client#assume_role_with_web_identity
+  #     ec2 = Aws::EC2::Client.new(credentials: role_credentials)
+  #
+  # If you omit `:client` option, a new {Aws::STS::Client} object will be
+  # constructed with additional options that were provided.
+  #
+  # Automatically handles refreshing credentials if an Expiration time is
+  # provided in the credentials payload.
   #
+  # This credential provider also provides a `before_refresh` callback
```

Comment:
I'm not sure we need to document this here - its already documented on the client initialize options.

Reply:
Assume Role Credentials had it already in the Class documentation (but that was the only instance). I can remove all of them, or keep all of them added here. Which do you prefer?

Reply:
I think it likely makes more sense to remove from class documentation and keep in initialize options.